xmpp.go

```diff
@@ -172,15 +185,12 @@ func (o Options) NewClient() (*Client, error) {
 		if o.TLSConfig != nil {
 			tlsconn = tls.Client(c, o.TLSConfig)
 		} else {
-			DefaultConfig.ServerName = strings.Split(o.User, "@")[1]
+			DefaultConfig.ServerName = host
```

Comment:
This was the culprit causing TLS verification errors.  We should be using the provided host name and not parsing it from the username, as they can (and do) differ from the host.